### ISSUE TITLE
Staging Review finding: Label for checkbox group not associated with checkboxes

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#87891]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** VA BTSSS - Travel Pay
**Product:** Travel Pay
**Feature:** MVP

## Findings details
**VA.gov Experience Standard - issue:** [User doesn't have enough information to complete a task.](https://design.va.gov/about/experience-standards/#user-has-enough-information-to-complete-a-task)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

In the Filter travel claims section, the grouping of checkboxes is labeled with "Travel claim status" to provide context for what each of the checkboxes is. However, that label isn't programmatically associated with the checkboxes.

Screen reader users rarely have their screen reader announce all of the text on the page, just as sighted users rarely read all of the text on the page and instead skim for the most useful parts. For screen reader users, that often means using a combination of in-page navigational techniques: navigating by landmarks, jumping heading-to-heading, and pressing [tab] to move between interactive elements.

When a screen reader user tabs to an interactive element, their screen reader will announce the element's type, label, and state -- eg. "checkbox, Submitted For Payment, checked." It only announces any other surrounding context if it's programmatically associated with that input.

As currently coded, "Travel claim status" is inside the same `fieldset` as the checkboxes, but it's coded as a `p`. Text inside a `fieldset` isn't automatically associated with the inputs in the `fieldset`. It's a fun quirk of HTML and the accessibility tree.

The intent of these checkboxes is still pretty clear without hearing "Travel claim status," so the impact is relatively small. But it is clarifying context that is probably better heard than not heard.

## Recommended action

The "Travel claim status" text should be in a `legend` within the `fieldset`:
```
<fieldset>
   <legend>Travel claim status</legend>
   <va-checkbox ... />
   ...
</fieldset>
```

The `legend` text can be styled to look like plain text here --- it's the programmatic association that matters. Now, when a screen reader user tabs to the first input in the `fieldset`, both the checkbox label and the legend text are announced.

Note that the `va-checkbox-group` component handles this for you out of the box, if you'd rather do that than putting `va-checkbox` components in your own `fieldset`. It also provides error handling for any error states that you might want to consider, like if no checkboxes are selected.

## References

- **Accessibility Defect Severity:** [3: Medium. Should be fixed in 1-3 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-3)
- **WCAG Success Criteria:** [WCAG 4.1.2 Name, Role, Value - Level A](https://www.w3.org/TR/WCAG21/#name-role-value)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.
